API/Controllers/ClientsController.cs

```diff
@@ -0,0 +1,42 @@
+using API.DTOs;
+using API.Entities;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class ClientsController(IOAuthClientService clientService) : Controller
+{
+    [HttpPatch("{id:int}/ratelimit")]
+    [Authorize(Roles = "admin")]
+    [EndpointSummary("Patches the ratelimit for a given client")]
+    public async Task<Results<BadRequest, NotFound, Ok<OAuthClientDTO>>> PatchRatelimitAsync(int id, [FromBody] JsonPatchDocument<RateLimitOverrides> patchedOverrides)
+    {
+        var overrides = new RateLimitOverrides();
```

Comment:
We should be pulling the client's existing overrides here instead of creating a new object. As written this gives no option for a partial update.

Reply:
Good point.